### PR TITLE
Feature/basic reservation event model with api

### PR DESCRIPTION
**Pull request by tomenet**
-Update app structure to MVP.
-User-reservation model and api extension.
-Integration with web.

### CODE REVIEW

File: tensorhive/controllers/reservation_event/CreateReservationEventController.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            return NoContent, 500
+
+        startTime = parsed_datetime(reservation_event['start'])
+        endTime = parsed_datetime(reservation_event['end'])
+        nodeId = reservation_event['nodeId']
+        userId = reservation_event['userId']
+        if (not UserModel.find_by_id(userId)):
+            return NoContent, 500
+
+        if (ReservationEventModel.find_events_the_same_time_nodes(startTime, endTime, nodeId) is not None):
```
Comment (micmarty) [inline tensorhive/controllers/reservation_event/CreateReservationEventController.py]: Maybe rename to:
```
find_node_events_between(start_time, end_time, node_id)???
find_node_colliding_events
find_events_with_time_collision
```

**Comment by tomenet** [comment]
I made changes. Pull request is ready!

File: tensorhive/api/api_specification.yml [hunk 2]
```diff
@@ -83,3 +117,38 @@ definitions:
...
+      # TODO Add user id, node id
+  ReservationEvent:
+    type: object
+    properties:
+      id:
+        type: integer
+        format: int64
+      title: { type: string }
+      description: { type: string }
+      nodeId: { type: integer}
```
Comment (roscisz) [inline tensorhive/api/api_specification.yml]: nodeId suggest a whole node, the reservation granularity should be per device.
Rename to deviceId? What if the reservation concerns more devices, do we want to create multiple events or one event with multiple devices in it?

File: tensorhive/controllers/reservation_event/CreateReservationEventController.py (hunk 1 above)
Comment (roscisz) [reply tensorhive/controllers/reservation_event/CreateReservationEventController.py]: I agree, but the same question as in api_specification: what is node, should it be a device?

File: tensorhive/api/api_specification.yml (hunk 2 above)
Comment (micmarty) [reply tensorhive/api/api_specification.yml]: We definitely must discuss this kind of dilemmas.

**Pull request merged**